cpp/src/parquet/stream_reader_test.cc

```diff
@@ -261,7 +275,176 @@ TEST_F(TestStreamReader, ValueChecking) {
     ASSERT_FLOAT_EQ(f, TestData::GetFloat(i));
     ASSERT_DOUBLE_EQ(d, TestData::GetDouble(i));
   }
+  ASSERT_EQ(reader_.current_row(), TestData::num_rows);
+  ASSERT_EQ(i, TestData::num_rows);
+}
+
+TEST_F(TestStreamReader, SkipRows) {
+  bool b;
+  std::string s;
+  std::array<char, 4> char_array;
+  char c;
+  int8_t int8;
+  uint16_t uint16;
+  int32_t int32;
+  uint64_t uint64;
+  std::chrono::microseconds ts_us;
+  float f;
+  double d;
+  std::string str;
+
+  const int num_rows_to_read = 3;
+  const int num_rows_to_skip = 13;
+  int num_rows_read = 0;
+  int i = 0;
+
+  // Skipping zero rows is ok.
+  //
+  ASSERT_EQ(0, reader_.SkipRows(0));
+
+  ASSERT_EQ(false, reader_.eof());
+  ASSERT_EQ(0, reader_.current_row());
+  ASSERT_EQ(TestData::num_rows, reader_.num_rows());
+
+  while (!reader_.eof()) {
+    for (int j = 0; !reader_.eof() && (j < num_rows_to_read); ++i, ++j) {
```

Comment:
what is i used for here?

Reply:
Just like the other tests, i is the current row being read.
I will add a comment to make this clear.